testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/suspendresume/DomainGracefulShutdownTestCase.java

```diff
@@ -309,7 +338,51 @@ public ModelNode call() throws Exception {
         }
     }
 
-    public static JavaArchive createDeployment() throws Exception {
+    @Test
+    public void testHostGracefulShutdownNoTimeout() throws Exception {
+        final String address = "http://" + TestSuiteEnvironment.getServerAddress() + ":8080/web-suspend";
+        final ExecutorService executorService = Executors.newFixedThreadPool(2);
+
+        Future<ModelNode> shutdownResult;
+        try {
+            executorService.submit(new Callable<Object>() {
+                @Override
+                public Object call() throws Exception {
+                    return HttpRequest.get(address, TimeoutUtil.adjust(60), TimeUnit.SECONDS);
+                }
+            });
+            TimeUnit.SECONDS.sleep(TimeoutUtil.adjust(1));
+
+            shutdownResult = executorService.submit(new Callable<ModelNode>() {
+                @Override
+                public ModelNode call() throws Exception {
+                    ModelNode op = new ModelNode();
+                    op.get(OP).set(SHUTDOWN);
+                    op.get(OP_ADDR).set(MASTER_ADDR.toModelNode());
+                    op.get(ModelDescriptionConstants.SUSPEND_TIMEOUT).set(0);
+                    op.get(ModelDescriptionConstants.RESTART).set(false);
+                    return domainMasterLifecycleUtil.executeAwaitConnectionClosed(op);
+                }
+            });
+
+            //nasty, but we need to make sure the server has time to stop before we check
+            TimeUnit.SECONDS.sleep(TimeoutUtil.adjust(2));
+
+            boolean timeoutExceptionOcurred = false;
+            try {
+                shutdownResult.get(TimeoutUtil.adjust(5), TimeUnit.SECONDS);
+            } catch (TimeoutException e) {
+                timeoutExceptionOcurred = true;
+            }
+            Assert.assertTrue(timeoutExceptionOcurred);
```

Comment:
I would assert false instead, that means the shutdown operation finished correctly, but there is an issue with this kind of tests using the Wildfly test suite. See my following comments.